src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1058,13 +1058,15 @@ def __getattribute__(self, name):
             return attr
         except AttributeError as ex:
             modified_search_results = []
-            for search_result in pyfluent.utils._search(
+            search_results = pyfluent.utils._search(
                 word=name, search_root=self, match_case=False, match_whole_word=False
-            ):
-                search_result = search_result.replace(
-                    "<search_root>", self.__class__.__name__
-                )
-                modified_search_results.append(search_result)
+            )
+            if search_results:
+                for search_result in search_results:
+                    search_result = search_result.replace(
+                        "<search_root>", self.__class__.__name__
+                    )
+                    modified_search_results.append(search_result)
```

Comment:
It would be better if we could move this code out of flobject.py. Might we be doing something similar for other APIs?

Reply:
Is it ok if w do this along with other refactoring as part of another PR? I wanted to merge this to fix the bug first.

Reply:
yes